Added: syntax highlighting, slurm docs and linux tutorial

th-blitz opened the pull request:
Hi, Added the following : 

1) Syntax highlighting for bash and console
2) Slurm documentation pages
3) Linux tutorial page
4) Hardware Specs page
5) Made changes to the structure and organization of the complete documentation

Note some links are dependent on other pages which needs to be created, and updated accordingly. Planning to update the links over the time as we create new pages and make changes.

Please let me know if any changes are required in these pages, or the documentation structure and I can send in another pull request.

th-blitz commented:
@s-sajid-ali Hi Sajid, 

I am working on a new PR as per linting warnings and to cleanup links.

You may provide feedback on documentation structure and formatting of the pages from this PR, in the meantime I will prepare the new one accordingly. 

Best,
Preetham

s-sajid-ali commented:
Any reason behind switching from `accessing-hpc` to `accessing_hpc`?

s-sajid-ali commented:
Most of the linter errors are for non-existent links to other pages in the documentation (`Tutorial 1` and `Hardware Specs` for instance). You can create empty pages for now and we can fill them in later. 

We can create issues for empty pages we create to keep track of the tasks we have delegated to the future to ensure that we don't miss filling them later.

th-blitz commented:
> Any reason behind switching from `accessing-hpc` to `accessing_hpc`?

yes, I decided to follow strict snake case naming convention going forward to make it easier in tracking the files.

th-blitz commented:
Hi @s-sajid-ali, 

Cleaned up the previous mess by squashing them to just 5 commits:

1. Added new pages and modified existing ones at `docs/` directory
2. Updated appropriate `prismThemes` for proper syntax highlighting of code blocks.
3.  Merged `slurm-docs` from `main` to  inherit the updated `pacakge.json` and `pnpm-lock` files.
4. Used `pnpm install` to add a new `remark-lint` package in-order to view linting suggestions locally which can be done by running the command `pnpm lint`.
5. Updated `pnpm-lock.yaml` file in the end, that was originally sourced from main to prevent generation of a differential lock file that may create merge conflicts.

If you have any questions please do let me know, and also feel free to indicate any changes that may be required.


Regards,
Preetham

s-sajid-ali commented:
Superseded by https://github.com/NYU-ITS/rts-docs-dev/pull/23, https://github.com/NYU-ITS/rts-docs-dev/pull/24
